cli/tests/program.rs

```diff
@@ -488,13 +488,27 @@ fn test_cli_program_deploy_with_authority() {
         program_data[..]
     );
 
-    // Set a new authority
+    // Set a new authority sign offline first
     let new_upgrade_authority = Keypair::new();
     config.signers = vec![&keypair, &upgrade_authority];
     config.command = CliCommand::Program(ProgramCliCommand::SetUpgradeAuthority {
         program_pubkey,
         upgrade_authority_index: Some(1),
         new_upgrade_authority: Some(new_upgrade_authority.pubkey()),
+        sign_only: true,
+        dump_transaction_message: false,
+    });
+    let sig_response = process_command(&config).unwrap();
+    let sign_only = parse_sign_only_reply_string(&sig_response);
+    let offline_pre_signer = sign_only.presigner_of(&keypair.pubkey()).unwrap();
+
+    config.signers = vec![&keypair, &upgrade_authority];
```

Comment:
To properly test the behavior, this should take the pre signer of the new upgrade authority, which is the offline signer
```suggestion
    let offline_pre_signer = sign_only.presigner_of(&new_upgrade_authority.pubkey()).unwrap();

    config.signers = vec![&keypair, &upgrade_authority, &offline_pre_signer];
```